### ISSUE TITLE
SDK ignores output type when targeting WinForms

### ISSUE THREAD

**Issue by davkean** [dotnet/sdk#15554]
For testing purposes, I often switch from a WinExe -> Exe so that I can see Console input. Unlike .NET Framework, I can't do that in SDK based projects, its ignoring my OutputType:"

``` XML
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0-windows</TargetFramework>
    <UseWindowsForms>true</UseWindowsForms>
  </PropertyGroup>

</Project>
```

It is understandable that OutputType defaults to WinExe if its not specified, but the project should always win out. This also breaks Properties pages that don't expect this at all.

**Comment by joeloff** [comment]
@davkean The SDK tries to infer the output type, but you can turn that off, e.g.

```
  <PropertyGroup>
    **<DisableWinExeOutputInference>true</DisableWinExeOutputInference>**
    <OutputType>Exe</OutputType>
  </PropertyGroup>
```

**Comment by davkean** [comment]
@joeloff The UI isn't aware of this, and the design that we've had with past properties is that the project file always wins.

**Comment by joeloff** [comment]
@davkean you're thinking property pages in VS where the user can just toggle between the "Windows Application" and "Console Application"?

**Comment by davkean** [comment]
Yes. I've been able to toggle WinForms apps between Console -> WinApp for coming on ~20 years, except for .NET Core.

**Comment by marcpopMSFT** [comment]
@dsplaisted please raise this in the partner meeting as we should consider what changes we want to make to the outputtype inference based on feedback we've gotten.

**Comment by ChahatKumar** [comment]
hi , i wanted to know how you exactly references the Windows.Forms dll in the .NET console app

When i try adding a COM reference , it gives error . Have you referenced using Object Browser ( under that in .net frameworks i found them )

apologies if this is out of topic here
